pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -1565,7 +1566,13 @@ func afwSchedulerEntry(client splcommon.ControllerClient, cr splcommon.MetaObjec
 	// while setting up the pipeline phases.
 	// Wait for the yield function to finish.
 	afwPipeline.phaseWaiter.Add(1)
-	go afwPipeline.afwYieldWatcher()
+
+	ctx, cancel := context.WithTimeout(
+		ctx,
+		time.Duration(afwPipeline.appDeployContext.AppFrameworkConfig.SchedulerYieldInterval)*time.Second,
+	)
+	defer cancel()
+	go afwPipeline.afwYieldWatcher(ctx)
```

Comment:
I am wondering if the same child context can be passed to all the phase managers, in which case that can be used as a termination signal.

Reply:
Yes, that is part of the next overall change in app framework code to use contexts. This is just an initial PR that introduces the context to this code. I will put the other change as an enhancement which requires some more code changes on the UTs too.

Reply:
Also, on a side note, the termination happens not only because of timeout but also if pipeline is empty. Hence, I think, using context.Done() only would not suffice.